### PR TITLE
Update s3 util

### PR DESCRIPTION
**Pull request by cpd85**
This will prevent an old bug in AWS SDK from occurring, where libcurl causes the program to crash

### CODE REVIEW

File: common/tests/s3_util_test.cpp [hunk 1]
```diff
@@ -51,11 +52,27 @@ TEST(S3UtilTest, ParseS3StringTest) {
   EXPECT_TRUE(std::get<1>(result) == "");
 }
 
+TEST(S3UtilTest, GetObjectS3Test) {
+  static shared_ptr<common::S3Util> myUtil =
+      common::S3Util::BuildS3Util(50, "media.pinterest.com", 60000, 60000);
```
Comment (shuzhang1989) [inline common/tests/s3_util_test.cpp]: let's remove this test.
it exposes some pinterest internal information.

File: common/s3util.h [hunk 2]
```diff
@@ -87,9 +89,12 @@ class S3Util {
   };
 
   // Don't recommend using this directly. Using BuildS3Util instead.
+  // If you must, make sure to call Aws::InitAPI(options); before constructor
   S3Util(const string& bucket,
-         const ClientConfiguration& client_config) :
-    bucket_(std::move(bucket)), s3Client(client_config) {
+         const ClientConfiguration& client_config,
+         SDKOptions options) :
```
Comment (shuzhang1989) [inline common/s3util.h]: const SDKOptions&

File: common/tests/s3_util_test.cpp (hunk 1 above)
Comment (shuzhang1989) [inline common/tests/s3_util_test.cpp]: also this test will be flaky, what if the s3 object is removed?

File: common/s3util.h [hunk 3]
```diff
@@ -36,6 +37,7 @@ using std::string;
 using std::vector;
 using std::tuple;
 using std::shared_ptr;
+using Aws::SDKOptions;
```
Comment (newpoo) [inline common/s3util.h]: nit: make it sorted?

File: common/s3util.h [hunk 4]
```diff
@@ -87,9 +89,12 @@ class S3Util {
   };
 
   // Don't recommend using this directly. Using BuildS3Util instead.
+  // If you must, make sure to call Aws::InitAPI(options); before constructor
```
Comment (newpoo) [inline common/s3util.h]: In this case, we can make it private?

File: common/s3util.cpp [hunk 5]
```diff
@@ -185,7 +185,9 @@ shared_ptr<S3Util> S3Util::BuildS3Util(
   aws_config.readRateLimiter = 
     std::make_shared<DefaultRateLimiter<>>(
           read_ratelimit_mb * 1024 * 1024);
-  return std::make_shared<S3Util>(bucket, aws_config);
+  SDKOptions options;
+  Aws::InitAPI(options);
```
Comment (newpoo) [inline common/s3util.cpp]: What happens if this gets called twice?

File: .travis.yml [hunk 6]
```diff
@@ -9,4 +9,4 @@ before_install:
 - docker pull aliced3645/rocksplicator-build
```
Comment (angxu) [inline .travis.yml]: change it to cdonaghy/rocksplicator-build?

**Pull request merged**